service/history/replicatorQueueProcessor.go

```diff
@@ -175,7 +175,7 @@ func (p *replicatorQueueProcessorImpl) readTasks(readLevel int64) ([]queueTaskIn
 		tasks[i] = response.Tasks[i]
 	}
 
-	return tasks, len(tasks) >= batchSize, nil
+	return tasks, len(response.NextPageToken) != 0, nil
```

Comment:
Looks like we don't use the nextPageToken for the next query.

Reply:
for anything other than timer, we use the task id for querying